Replace `logging.SlogNopHandler` by `slog.DiscardHandler`

tklauser opened the pull request:
The `log/slog` Go standard library package provides `DiscardHandler` since Go 1.24[^1]. Make use of it instead of duplicating it as `logging.SlogNopHandler`.

[^1]: https://go.dev/doc/go1.24#logslogpkglogslog

sayboras reviewed (approved):
it's happening go.go go go go go ... (24 times)

derailed reviewed (approved):
@tklauser Nice! Thanks for this update.

tklauser committed:
Replace logging.SlogNopHandler by slog.DiscardHandler

The log/slog Go standard library package provides DiscardHandler since
Go 1.24[^1]. Make use of it instead of duplicating it as
logging.SlogNopHandler.

[^1]: https://go.dev/doc/go1.24#logslogpkglogslog

Signed-off-by: Tobias Klauser <tobias@cilium.io>
